### PR TITLE
Fix expected error text

### PR DESCRIPTION
**Pull request by ncdc**
I had initially coded the expected text to match what was coming out of stderr, which was wrong. This makes it match against what comes from stdout.

### CODE REVIEW

**Comment by ncdc** [comment]
@k8s-bot node e2e test this

File: hack/make-rules/test-cmd-util.sh [hunk 1]
```diff
@@ -1064,11 +1064,11 @@ run_kubectl_get_tests() {
 
   ## check --allow-missing-template-keys=false results in an error for a missing key with jsonpath
   output_message=$(! kubectl get pod valid-pod --allow-missing-template-keys=false -o jsonpath='{.missing}' "${kube_flags[@]}")
```
Comment (ixdy) [inline hack/make-rules/test-cmd-util.sh]: alternately, would it be better to capture stderr?

**Comment by ncdc** [comment]
I could do that. Either way works for me.

**Comment by fejta** [comment]
ref https://github.com/kubernetes/test-infra/issues/1561
@k8s-bot cri node e2e test this
@k8s-bot node e2e test this

**Comment by ixdy** [comment]
@ncdc yeah, I don't really have a strong preference either way.

if this correctly tests as-written, then
/lgtm

**Pull request merged**